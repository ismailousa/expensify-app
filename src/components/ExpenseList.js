import React from 'react';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';
import { connect } from 'react-redux';

export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (
                <p>No Expenses</p>
            ) : (
            props.expenses.map((expense) => ( <ExpenseListItem key={expense.id} {...expense} />))
            )
        }
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseList);