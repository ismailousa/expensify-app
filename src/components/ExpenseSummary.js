import React from 'react';
import selectVisibleExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import { connect } from 'react-redux';
import numeral from 'numeral';

export const ExpenseSummary = ({expenseCount, expensesTotal}) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    return (
    <div>
        <p>Viewing {expenseCount} {expenseWord} totalling {numeral(expensesTotal / 100).format('$0,0.00')}</p>
    </div>
    )
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectVisibleExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpenseSummary);

