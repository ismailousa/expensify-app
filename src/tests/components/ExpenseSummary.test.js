import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';

test('Should render correctly for 1 expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expensesTotal={255} />);
    expect(wrapper).toMatchSnapshot();
});

test('Should render correctly for multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={5} expensesTotal={255600}/>);
    expect(wrapper).toMatchSnapshot();
});