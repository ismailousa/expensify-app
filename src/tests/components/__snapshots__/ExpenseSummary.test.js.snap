// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly for 1 expenses 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totalling 
    $2.55
  </p>
</div>
`;

exports[`Should render correctly for multiple expenses 1`] = `
<div>
  <p>
    Viewing 
    5
     
    expenses
     totalling 
    $2,556.00
  </p>
</div>
`;
